flexmeasures/data/tests/test_scheduling_repeated_jobs.py

```diff
@@ -0,0 +1,206 @@
+from datetime import datetime, timedelta
+import copy
+import logging
+
+import pytz
+import pytest
+
+from flexmeasures.data.models.data_sources import DataSource
+from flexmeasures.data.models.time_series import Sensor
+from flexmeasures.data.tests.utils import work_on_rq, exception_reporter
+from flexmeasures.data.services.scheduling import create_scheduling_job
+from flexmeasures.data.services.utils import hash_function_arguments
+
+
+@pytest.mark.parametrize(
+    "args_modified,kwargs_modified,equal",
+    [
+        (
+            [1, 2, "1"],
```

Comment:
What if you change the order of the args? For `["1", 2, 1]` I'd expect `equal=False`.
What if you change the order of the contents of a nested dictionary? For `"key4": {"key2_nested": 2, "key1_nested": 1}` I'd expect `equal=True`, but I could also live with a `False`. Just good to know what the behavior is.

I think our job's dict-like variables (e.g. "flex_model" and "flex_context") might actually still be stored as JSON strings when they are passed off to the job, so if a user swaps the order of their contents that would probably lead to invoking a new scheduling job.

Reply:
Not asking to adjust the hashing logic, just asking to add test cases so we get a full understanding of the behavior.

Reply:
Good edge cases, I add them all. Dictionaries are sorted by key and then hashed, which makes order invariant hashes with respect to keys.

Reply:
That's great!

For completeness, can you add the serialized dict case, too, with a note regarding the implications (see my previous comment)? That is, I expect the following to not produce the same hash right now.
```
"key5": '{"serialized_key1_nested": 1, "serialized_key2_nested": 2}'
```
```
"key5": '{"serialized_key2_nested": 2, "serialized_key1_nested": 1}'
```